### PR TITLE
[KYUUBI #983] Refactor the basic abstraction of frontend service

### PR DESCRIPTION
**Pull request by yanghua**
…

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/service/ThriftFrontendService.scala [hunk 1]
```diff
@@ -105,11 +105,11 @@ class ThriftFrontendService private(name: String, be: BackendService, oomHook: R
     super.initialize(conf)
   }
 
-  override def connectionUrl(server: Boolean = false): String = {
+  override def connectionUrl: String = {
     getServiceState match {
       case s @ ServiceState.LATENT => throw new IllegalStateException(s"Illegal Service State: $s")
       case _ =>
-        if (server || conf.get(ENGINE_CONNECTION_URL_USE_HOSTNAME)) {
+        if (conf.get(ENGINE_CONNECTION_URL_USE_HOSTNAME)) {
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/service/ThriftFrontendService.scala:112]: this is an engine config, we need to create `SparkSQLFrontendService` to remove  the `server` flag

Comment (yanghua) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/service/ThriftFrontendService.scala:112]: Did you mean this PR introduce the side-effort? When diving into the code base, I found the server flag only be used at [here](https://github.com/apache/incubator-kyuubi/blob/master/kyuubi-common/src/main/scala/org/apache/kyuubi/service/ThriftFrontendService.scala#L131), right? Before this PR, the same logic is also been judged in `KyuubiServer`(KyuubiServer used ThriftFrontendService).

In fact, I haven't thought about the real role of SparkSQLFrontendService. Will it be a CompositeService? ThriftFrontendService will still be used by KyuubiServer. You mean we need to put the judgment logic of the engine configuration in CompositeService (SparkSQLFrontendService)?

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiFrontendService.scala [hunk 2]
```diff
@@ -19,30 +19,30 @@ package org.apache.kyuubi.server
...
 import org.apache.kyuubi.Logging
 import org.apache.kyuubi.config.KyuubiConf
-import org.apache.kyuubi.service.{AbstractFrontendService, BackendService, ServiceState, ThriftFrontendService}
+import org.apache.kyuubi.service.{AbstractFrontendService, BackendService, CompositeService, ServiceState, ThriftFrontendService}
 
 /**
  * A kyuubi frontend service is a kind of composite service, which used to
  * composite multiple frontend services for kyuubi server.
  */
 class KyuubiFrontendService private(name: String, be: BackendService)
```
[the remark this answers is not in the document]
Comment (yanghua) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiFrontendService.scala]: addressed

**Review by yaooqinn — APPROVED** [review]

**Comment by yanghua** [comment]
@yaooqinn Can we merge this PR so that I can apply it in my local?

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**